pmlc/conversion/comp_to_llvm/comp_to_vulkan.cc

```diff
@@ -0,0 +1,472 @@
+  static constexpr const char *kBindBufferFloat16 = "bindBufferFloat16";
+  static constexpr const char *kBindBufferFloat32 = "bindBufferFloat32";
+  static constexpr const char *kBindBufferFloat64 = "bindBufferFloat64";
+  static constexpr const char *kBindBufferInteger8 = "bindBufferInteger8";
+  static constexpr const char *kBindBufferInteger16 = "bindBufferInteger16";
+  static constexpr const char *kBindBufferInteger32 = "bindBufferInteger32";
+  static constexpr const char *kBindBufferInteger64 = "bindBufferInteger64";
+
+  const char *getBufferBindingFunc(mlir::Type elementType) {
+    if (elementType.isInteger(8)) {
+      return kBindBufferInteger8;
+    }
+    if (elementType.isInteger(16)) {
+      return kBindBufferInteger16;
+    }
+    if (elementType.isInteger(32)) {
+      return kBindBufferInteger32;
+    }
+    if (elementType.isInteger(64)) {
+      return kBindBufferInteger64;
+    }
+    if (elementType.isBF16()) {
+      return kBindBufferBFloat16;
+    }
+    if (elementType.isF16()) {
+      return kBindBufferFloat16;
+    }
+    if (elementType.isF32()) {
+      return kBindBufferFloat32;
+    }
+    if (elementType.isF64()) {
+      return kBindBufferFloat64;
+    }
+    return nullptr;
+  }
+};
+
+struct ConvertScheduleFunc : ConvertCompToVkBasePattern<comp::ScheduleFunc> {
+  ConvertScheduleFunc(const BinaryModulesMap &modulesMap,
+                      mlir::TypeConverter &typeConverter,
+                      mlir::MLIRContext *context)
+      : ConvertCompToVkBasePattern<comp::ScheduleFunc>(typeConverter, context),
+        modulesMap(modulesMap) {}
+
+  mlir::LogicalResult
+  matchAndRewrite(comp::ScheduleFunc op, mlir::ArrayRef<mlir::Value> operands,
+                  mlir::ConversionPatternRewriter &rewriter) const override;
+
+  const BinaryModulesMap &modulesMap;
+};
+
+} // namespace
+
+void populateCompToVkPatterns(mlir::MLIRContext *context,
+                              const BinaryModulesMap &modulesMap,
+                              mlir::TypeConverter &typeConverter,
+                              mlir::OwningRewritePatternList &patterns) {
+  // Populate type conversion patterns.
+  LLVM::LLVMType llvmInt8Ptr = LLVM::LLVMType::getInt8PtrTy(context);
+  typeConverter.addConversion(
+      [=](comp::ExecEnvType execEnvType) -> mlir::Optional<mlir::Type> {
+        return llvmInt8Ptr;
+      });
+  typeConverter.addConversion(
+      [=](comp::EventType eventType) -> mlir::Optional<mlir::Type> {
+        return llvmInt8Ptr;
+      });
+  patterns.insert<ConvertToInitVulkan>(kInitVulkan, typeConverter, context);
+  patterns.insert<ConvertToDeinitVulkan>(kDeinitVulkan, typeConverter, context);
+  patterns.insert<ConvertToCommandBufferSubmit>(kRun, typeConverter, context);
+
+  patterns.insert<ConvertScheduleFunc>(modulesMap, typeConverter, context);
+  patterns.insert<ConvertAlloc>(typeConverter, context);
+
+  // TODO modify those to vulkan api
```

Comment:
Rob should soon have a new revision for [RFC:-Network-Pre-Initializationl](https://github.com/plaidml/plaidml/wiki/RFC:-Network-Pre-Initializationl), we should refer this design.